File: oj-web/src/enums/system/meta/ImageBusinessType.ts
import type { BaseEnum } from '@/enums/base';

// 图片业务类型枚举接口
interface ImageBusinessTypeEnum extends BaseEnum {
}

/**
 * 图片业务类型枚举
 * 用于标识不同的图片业务类型
 */
export const ImageBusinessType: { [key: string]: ImageBusinessTypeEnum } = {
  /**
   * 轮播图
   */
  CAROUSEL: {
    code: 'Carousel',
    name: '轮播图'
  },

  /**
   * 背景图
   */
  BACKGROUND: {
    code: 'Background',
    name: '背景图'
  },

  /**
   * 头像
   */
  AVATAR: {
    code: 'Avatar',
    name: '头像'
  },

  /**
   * 普通图片
   */
  NORMAL_IMAGE: {
    code: 'NormalImage',
    name: '普通图片'
  },

  /**
   * 临时图片
   */
  TEMP: {
    code: 'Temp',
    name: '临时图片'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): ImageBusinessTypeEnum | null {
  return Object.values(ImageBusinessType).find(item => item.code === code) || null;
} 

File: glowxq-api/business/business-system/src/main/java/com/glowxq/system/meta/enums/ImageBusinessType.java
package com.glowxq.system.meta.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.system.meta.base.ImageBusinessEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum ImageBusinessType implements BaseType, ImageBusinessEnum {

    Carousel("HomeImage", "轮播图"),
    Background("Background", "背景图片"),
    NormalImage("NormalImage", "普通图片"),
    Avatar("Avatar", "头像"),
    Temp("Temp", "临时图片"),

    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static ImageBusinessType matchCode(String code) {
        for (ImageBusinessType pushStatus : ImageBusinessType.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}
